tests/test_api.py

```diff
@@ -316,6 +316,24 @@ def test_searchcv_init_params(self, config):
             loader,
         )
 
+    @pytest.mark.parametrize(
+        "config", ["MLPClassifier"],
+    )
+    def test_searchcv_routed_params(self, config):
+        """Tests compatibility with Scikit-learn's hyperparameter search CV."""
+        loader, model, build_fn, _ = CONFIG[config]
+        estimator = model(build_fn, epochs=1, hidden_layer_sizes=[])
+        params = {
+            "build__hidden_layer_sizes": [[], [5]],
+            "compile__optimizer": ["sgd", "adam"],
+        }
+        search = GridSearchCV(estimator, params)
+        basic_checks(search, loader)
+        assert search.best_estimator_.model_.optimizer._name.lower() in (
+            "sgd",
+            "adam",
+        )
+
```

Comment:
@stsievert does this test satisfy your use case in https://github.com/adriangb/scikeras/issues/50#issuecomment-680179195?

Reply:
This is specifically testing with routed parameters. The test above uses `epochs`, `validation_split` and `hidden_layer_sizes`, which are non-routed parameters.

Reply:
This test is specifically testing routed params. The test above (`test_searchcv_init_params`) is the same but for unrouted params. I added the `optimizer` parameter to `test_searchcv_init_params` for completeness.